src/core/core.controller.js

```diff
@@ -17,38 +17,62 @@ import {version} from '../../package.json';
 
 const valueOrDefault = helpers.valueOrDefault;
 
+function getIndexAxis(type, options) {
+	const typeDefaults = defaults[type] || {};
+	const datasetDefaults = typeDefaults.datasets || {};
+	const typeOptions = options[type] || {};
+	const datasetOptions = typeOptions.datasets || {};
+	return datasetOptions.indexAxis || options.indexAxis || datasetDefaults.indexAxis || 'x';
+}
+
+function getAxisFromDefaultScaleID(id, indexAxis) {
+	let axis = id;
+	if (id === '_index_') {
+		axis = indexAxis;
+	} else if (id === '_value_') {
+		axis = indexAxis === 'x' ? 'y' : 'x';
+	}
+	return axis;
+}
+
+function getDefaultScaleIDFromAxis(axis, indexAxis) {
+	return axis === indexAxis ? '_index_' : '_value_';
+}
+
 function mergeScaleConfig(config, options) {
 	options = options || {};
 	const chartDefaults = defaults[config.type] || {scales: {}};
 	const configScales = options.scales || {};
+	const chartIndexAxis = getIndexAxis(config.type, options);
 	const firstIDs = {};
 	const scales = {};
 
 	// First figure out first scale id's per axis.
-	// Note: for now, axis is determined from first letter of scale id!
 	Object.keys(configScales).forEach(id => {
-		const axis = id[0];
+		const scaleConf = configScales[id];
+		const axis = scaleConf.axis || id[0];
+		const defaultId = getDefaultScaleIDFromAxis(axis, chartIndexAxis);
 		firstIDs[axis] = firstIDs[axis] || id;
-		scales[id] = helpers.mergeIf({}, [configScales[id], chartDefaults.scales[axis]]);
+		scales[id] = helpers.mergeIf({axis}, [scaleConf, chartDefaults.scales[axis], chartDefaults.scales[defaultId]]);
 	});
 
 	// Backward compatibility
 	if (options.scale) {
-		scales[options.scale.id || 'r'] = helpers.mergeIf({}, [options.scale, chartDefaults.scales.r]);
+		scales[options.scale.id || 'r'] = helpers.mergeIf({axis: 'r'}, [options.scale, chartDefaults.scales.r]);
 		firstIDs.r = firstIDs.r || options.scale.id || 'r';
 	}
 
 	// Then merge dataset defaults to scale configs
 	config.data.datasets.forEach(dataset => {
-		const datasetDefaults = defaults[dataset.type || config.type] || {scales: {}};
+		const type = dataset.type || config.type;
+		const indexAxis = dataset.indexAxis || getIndexAxis(type, options);
+		const datasetDefaults = defaults[type] || {};
 		const defaultScaleOptions = datasetDefaults.scales || {};
 		Object.keys(defaultScaleOptions).forEach(defaultID => {
-			const id = dataset[defaultID + 'AxisID'] || firstIDs[defaultID] || defaultID;
+			const axis = getAxisFromDefaultScaleID(defaultID, indexAxis);
+			const id = dataset[axis + 'AxisID'] || firstIDs[axis] || axis;
 			scales[id] = scales[id] || {};
-			helpers.mergeIf(scales[id], [
-				configScales[id],
-				defaultScaleOptions[defaultID]
-			]);
+			helpers.mergeIf(scales[id], [{axis}, configScales[id], defaultScaleOptions[defaultID]]);
```

Comment:
I think it might be a bit confusing that there's two ways to do things now. I wonder if we should also replace `x` and `y` configs in the other chart types by `index` and `value`. E.g. if you wanted to set `indexAxis` on a `bubble` or `line` dataset it wouldn't necessarily just work right now because we only updated the defaults for the `BarController`. I think it'd be clearer how things worked if we did that universally. And in that case we probably wouldn't want the underscores because I think they're options that users would be setting more frequently

Reply:
Those controllers use same defaults for both axes, so makes no difference. Its also likely that some (external) chart types want to use some defaults for x and others for y regardless of the indexAxis.

I think you misunderstand how this works. Users are not supposed to use '_index_' and '_value_' in their configuration. They use 'x' and 'y' (or whatever they like) and the defaults from the '_index_' placeholder are applied to the scale that is on the axis marked as `indexAxis`. And the '_value_' defaults are applied to the other scale.

Reply:
I was thinking it might be nice if they could use `index` though. Imagine if you wanted to build a chart builder like the dialog to insert a chart in Excel or Google Spreadsheets. You could just toggle back and forth between horizontal and vertical by changing only a single property. Otherwise you need to move all the properties between `x` and `y`. I think it'd be fairly messy without using `_index_` and `_value_` to make a bar chart that can toggle between horizontal and vertical

Reply:
I see. So we'd require the scales to be defined as `index` and `value`. What if the user needs multiple value scales?

Add `iScaleID` and `vScaleID`? then the user would be free to name the scales again, but we still need to match the defaults for correct scales.

Reply:
Yeah, you've raised some good points. Maybe we should just leave it the way it is.

Though I do still wonder if we should `_index_` and `_value_` in the other controllers. You said the other controllers use the same defaults for both axes, but I'm not sure that's true. The `line` controller has a `line` y-axis and `category` x-axis, right? Maybe it should be `line` for `_value_` and `category` for `_index_` instead?